hibernate-core/src/main/java/org/hibernate/dialect/HSQLDialect.java

```diff
@@ -313,78 +313,55 @@ public String getFromDual() {
 	}
 
 	@Override
-	public ViolatedConstraintNameExtracter getViolatedConstraintNameExtracter() {
-		return version < 200 ? EXTRACTER_18 : EXTRACTER_20;
-	}
-
-	private static final ViolatedConstraintNameExtracter EXTRACTER_18 = new TemplatedViolatedConstraintNameExtracter() {
-		@Override
-		protected String doExtractConstraintName(SQLException sqle) throws NumberFormatException {
-
-			final int errorCode = JdbcExceptionHelper.extractErrorCode( sqle );
-
-			switch (errorCode) {
-				case -8:
-					return extractUsingTemplate(
-							"Integrity constraint violation ", " table:",
-							sqle.getMessage()
-					);
-				case -9:
-					return extractUsingTemplate(
-							"Violation of unique index: ", " in statement [",
-							sqle.getMessage()
-					);
-				case -104:
-					return extractUsingTemplate(
-							"Unique constraint violation: ", " in statement [",
-							sqle.getMessage()
-					);
-				case -177:
-					return extractUsingTemplate(
-							"Integrity constraint violation - no parent ", " table:",
-							sqle.getMessage()
-					);
-			}
-			return null;
-		}
-
-	};
+	public ViolatedConstraintNameExtractor getViolatedConstraintNameExtracter() {
+		return version < 200 ? EXTRACTOR_18 : EXTRACTOR_20;
+	}
+
+	private static final ViolatedConstraintNameExtractor EXTRACTOR_18 =
+			new TemplatedViolatedConstraintNameExtractor( sqle -> {
+				switch ( JdbcExceptionHelper.extractErrorCode( sqle ) ) {
+					case -8:
+						return extractUsingTemplate(
+								"Integrity constraint violation ", " table:",
+								sqle.getMessage()
+						);
+					case -9:
+						return extractUsingTemplate(
+								"Violation of unique index: ", " in statement [",
+								sqle.getMessage()
+						);
+					case -104:
+						return extractUsingTemplate(
+								"Unique constraint violation: ", " in statement [",
+								sqle.getMessage()
+						);
+					case -177:
+						return extractUsingTemplate(
+								"Integrity constraint violation - no parent ", " table:",
```

Comment:
Could we add a space char after 'table:' to be consistent with #359 below?

Reply:
Look, I mean, in principle I shouldn't really be changing legacy code like that without testing my changes. Why knows what inconsistently-formatted error messages HSQL might produce?

Reply:
You decide. After my raising awareness, I am gone, :).

Reply:
Not going to change it. :-)